Mathlib/Analysis/Analytic/Constructions.lean

```diff
@@ -1174,7 +1174,17 @@ theorem Finset.analyticWithinAt_prod {A : Type*} [NormedCommRing A] [NormedAlgeb
 
 /-- Finite products of analytic functions are analytic -/
 @[fun_prop]
-theorem Finset.analyticAt_prod {A : Type*} [NormedCommRing A] [NormedAlgebra 𝕜 A]
+theorem Finset.analyticAt_prod {α : Type*} {A : Type*} [NormedCommRing A] [NormedAlgebra 𝕜 A]
+    {f : α → E → A} {c : E} (N : Finset α) (h : ∀ n ∈ N, AnalyticAt 𝕜 (f n) c) :
+    AnalyticAt 𝕜 (∏ n ∈ N, f n) c := by
+  simp_rw [← analyticWithinAt_univ] at h ⊢
+  have : (∏ n ∈ N, f n) = (fun z ↦ ∏ n ∈ N, f n z) := by aesop
+  rw [this]
+  exact N.analyticWithinAt_prod h
```

Comment:
If you move this one after the `fun_prod` version, it has a simpler proof, like
```
  convert N.analyticAt_fun_prod h
  simp
```

Reply:
Implemented.